### DISCUSSION TITLE
cluster startup with couchbase gets 401 but values confirmed

### DISCUSSION THREAD

**Discussion by ksully-execute** [JanssenProject/jans#6892] [Q&A] [answered]
Using the github jans values file, altered for my existing cluster, and using couchbase.
Have the values adjusted, and verified that the superuser and restricted user values are correctly defined, and confirmed access through curl using the cnCouchbaseUrl and both cnCouchbaseUser/cncnCouchbasePassword and cnCouchbaseSuperUser/cnCouchbaseSuperUserPassword. The value work (converted to a basic auth header).
So why am I seeing messages like this?
` WARNING - jans.pycloudlib.lock - 2023-12-01 19:50:08,275 - Backing off LockManager.check_connection(<LockRecord(adapter=CouchbaseLock)>) for 10.0 seconds; error='requests.exceptions.HTTPError: 40 │
│ 1 Client Error: Unauthorized for url: https://couchbasedev-0000.couchbase.svc.cluster.local:18091/pools/default/buckets'   `

`│ WARNING - jans.pycloudlib.lock - 2023-12-01 19:50:28,123 - Giving up LockManager.check_connection(<LockRecord(adapter=CouchbaseLock)>) after 7 tries within 60.1 seconds; error='requests.exception │
│ s.HTTPError: 401 Client Error: Unauthorized for url: https://couchbasedev-0000.couchbase.svc.cluster.local:18091/pools/default/buckets'  `

and of course other components saying this, which I assume is because of the above?
`WARNING - jans.pycloudlib.wait - 2023-12-01 19:53:23,671 - Config is not ready; reason=Config 'hostname' is not available; retrying in 10.0 seconds  `
Thanks for any thoughts

**Comment by ossdhaval** [comment]
Adding @misba7 for comments.

**Comment by misba7** [comment]
`Config is not ready; reason=Config 'hostname' is not available` shows that the config job didn't complete successfully. Since the config job populates `hostname` and other data in the configmap named `cn`, which didn't happen in your case.
This will subsequently cause the persistence job to fail too, and then the rest of the pods.
So try redeploying, and keep an eye on the config-job.

**Reply by ksully-execute** [reply]
Thanks @misba7 , I have watched the config job before, and the first thing that shows up is that warning. Is there a way to get further info (turn up debug or something)? 
I've verified the couchbase credentials (cnCouchbaseUser/cnCouchbasePassword and cnCoubhaseSuperUser/cnCouchbaseSuperUserPassword) that I have the in yaml can access the cnCouchbaseURL that I have in there
 cnCouchbaseUrl: couchbasedev-0000.couchbase.svc.cluster.local
So I'm not sure why it's getting a 401...any thoughts?

**Reply by misba7** [reply, marked as answer]
No, this warning is from other components, stating that the config-job pod didn't populate the configmap cn with  its data(hostname being part of the data).

So that's why I am saying try to delete the jans namespace, do a fresh install and check the config-job pod.

**Comment by ksully-execute** [comment]


**Reply by ksully-execute** [reply]
After deleting the instance, and the namespace, here's what I see in the config pods:

` INFO - jans.pycloudlib.wait - 2023-12-02 13:29:37,780 - Config is ready                                                                                                                             
 INFO - jans.pycloudlib.wait - 2023-12-02 13:29:37,794 - Secret is ready                                                                                                                             
 WARNING - jans.pycloudlib.lock - 2023-12-02 13:29:37,882 - Backing off LockManager.check_connection(<LockRecord(adapter=CouchbaseLock)>) for 10.0 seconds; error='requests.exceptions.HTTPError: 40 
 1 Client Error: Unauthorized for url: https://couchbasedev-0000.couchbase.svc.cluster.local:18091/pools/default/buckets'                                                                            


And in a second config job


 WARNING - jans.pycloudlib.lock - 2023-12-02 13:33:09,700 - Giving up LockManager.check_connection(<LockRecord(adapter=CouchbaseLock)>) after 7 tries within 60.0 seconds; error='requests.exception 
 Traceback (most recent call last):                                                                                                                                                                  
   File "/app/scripts/bootstrap.py", line 612, in <module>                                                                                                                                           
     cli(prog_name="configurator")                                                                                                                                                                   
   File "/usr/lib/python3.11/site-packages/click/core.py", line 722, in __call__                                                                                                                     
     return self.main(*args, **kwargs)                                                                                                                                                               
            ^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                               
   File "/usr/lib/python3.11/site-packages/click/core.py", line 697, in main                                                                                                                         
     rv = self.invoke(ctx)                                                                                                                                                                           
          ^^^^^^^^^^^^^^^^                                                                                                                                                                           
   File "/usr/lib/python3.11/site-packages/click/core.py", line 1066, in invoke                                                                                                                      
     return _process_result(sub_ctx.command.invoke(sub_ctx))                                                                                                                                         
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                          
   File "/usr/lib/python3.11/site-packages/click/core.py", line 895, in invoke                                                                                                                       
     return ctx.invoke(self.callback, **ctx.params)                                                                                                                                                  
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                  
   File "/usr/lib/python3.11/site-packages/click/core.py", line 535, in invoke                                                                                                                       
     return callback(*args, **kwargs)                                                                                                                                                                
            ^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                
   File "/app/scripts/bootstrap.py", line 557, in load                                                                                                                                               
     with manager.lock.create_lock("configurator-load"):                                                                                                                                             
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                              
   File "/usr/lib/python3.11/site-packages/jans/pycloudlib/lock/__init__.py", line 158, in create_lock                                                                                               
     self.check_connection(lock)                                                                                                                                                                     
   File "/usr/lib/python3.11/site-packages/backoff/_sync.py", line 105, in retry                                                                                                                     
     ret = target(*args, **kwargs)                                                                                                                                                                   
           ^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                   
   File "/usr/lib/python3.11/site-packages/jans/pycloudlib/lock/__init__.py", line 112, in check_connection                                                                                          
     connected = lock.adapter.connected()                                                                                                                                                            
                 ^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                            
   File "/usr/lib/python3.11/site-packages/jans/pycloudlib/lock/couchbase_lock.py", line 178, in connected                                                                                           
     self._prepare_bucket()                                                                                                                                                                          
   File "/usr/lib/python3.11/site-packages/jans/pycloudlib/lock/couchbase_lock.py", line 219, in _prepare_bucket                                                                                     
     resp.raise_for_status()                                                                                                                                                                         
   File "/usr/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status                                                                                                       
     raise HTTPError(http_error_msg, response=self)                                                                                                                                                  
 requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://couchbasedev-0000.couchbase.svc.cluster.local:18091/pools/default/buckets   `

**Reply by misba7** [reply]
That seems like a Couchbase config error rather than Jans.

If you are sure that the supplied credentials have the needed permissions, then maybe `override.yaml` is not supplied correctly. 

Try `helm install janssen janssen/janssen -n jans -f override.yaml --dry-run --debug | grep Couchbase` and double-check if these are the Couchbase credentials you intend to supply. 
If not, fix your `override.yaml`

**Reply by ksully-execute** [reply]
(appreciate the help!)
the values  look right, and when combined for basic Auth header, they connect fine with curl going to that same address that's specified in cnCouchbaseUrl....
Do you think the cnCouchbaesCrt might be part of this? I don't know that it would cause a 401 though....there are multiple certificates in couchbase (couchbase server, couchbase-cluster-ca adn node certificates). I've tried the server and cluster and it doesn't seem to make a difference.

If not that, then is there further detail I could get, like see what Basic Auth header the code is producing to compare with what works for me?

**Reply by misba7** [reply]
Yes it can be. 
We discussed the `cnCouchbaesCrt ` and how to pass it in a previous [discussion](https://github.com/JanssenProject/jans/discussions/5589#discussioncomment-6517805) and it seemed that it worked with you

**Reply by ksully-execute** [reply]
also, just to confirm (not a k8s expert)
When I check the credentials against the couchbase pod, the name in chCouchbaseUrl isn't resolved from the command line of a control plane machine (which I believe is normal), but is resolvable through a dns lookup within a pod, like this:
`
Server:         10.96.0.10
Address:        10.96.0.10#53

Name:   couchbasedev-0000.couchbase.svc.cluster.local
Address: 10.111.32.72
`

And then using that 10.111.32.71 address, the curl command like this work fine:
`curl  -k --location https://10.111.32.72:18091/pools/default/buckets --header Content-Type: application/x-www-form-urlencoded --header 'Authorization: Basic ZGGV2MTpxZXdrajBwcXc5Z2EwOTRnOXF3aDRnYXdoZ2hQQVc0ODk='

 curl  -k --location https://10.111.32.72:18091/pools/default/buckets --header Content-Type: application/x-www-form-urlencoded --header 'Authorization: Basic ZGV2MTpxZXdrajBwcXc5Z2EwOTRnOXF3aDRnYXdoZ2hQQVc0ODk='
[{"name":"travel-sample","nodeLocator":"vbucket","bucketType":"membase","storageBackend":"couchstore","uuid":"f1da38c8faf3258a81047d293989b460","uri":"/pools/default/buckets/travel-sample?bucket_uuid=f1da38c8faf3258a81047d293989b460","streamingUri":"/pools/default/bucketsStreaming/travel-sample?bucket_uuid=f1da38c8faf3258a81047d293989b460","bucketCapabilitiesVer":"","bucketCapabilities":["collections","durableWrite","tombstonedUserXAttrs","couchapi","subdoc.ReplaceBodyWithXattr","subdoc.DocumentMacroSupport","subdoc.ReviveDocument","dcp.IgnorePurgedTombstones","dcp","cbhello","touch","cccp","xdcrCheckpointing","nodesExt","xattr"],"collectionsManifestUid":"1","ddocs":{"uri":"/pools/default/buckets/travel-sample/ddocs"},"vBucketServerMap":{"hashAlgorithm":"CRC","numReplicas":0,"serverList":["couchbasedev-0000.couchbasedev.couchbase.svc:11210"],"vBucketMap":[[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[0],[
`

**Reply by ksully-execute** [reply]
thanks for the point on the cert. My memory is not so good, appreciate the pointer to the prior discussion. Will change that

**Reply by ksully-execute** [reply]
cert done, and put testenvironment to true as well, no joy, same error on couchbase. 
I guess I can try go get mysql or something just to get past this for now (couchbase being my persistence of choice).
I'll take any other suggestions though (see 2 posts up for notes on the verification on couchbase creds).
Thanks

**Reply by ksully-execute** [reply]
is there a way to get more debug into, specifically the basic auth head that it's trying to use?
Or perhaps a way to get the URL used for couchbase to be http instead of https?

**Reply by misba7** [reply]
There's `appLoggers` to configure log level for every component. For example you can configure `global.auth-server.appLoggers` [here](https://github.com/JanssenProject/jans/blob/c49f8f16af502429bf7a58fc31f6d5ffc1f67f78/charts/janssen/values.yaml#L662C1-L692C33)
